plugin/pkg/admission/defaultpodnetworknotready/admission.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2020 Authors of Arktos.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package defaultpodnetworknotready
+
+import (
+	"io"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apiserver/pkg/admission"
+	api "k8s.io/kubernetes/pkg/apis/core"
+)
+
+const (
+	// PluginName indicates name of admission plugin.
+	PluginName = "DefaultPodNetworkNotReady"
+
+	networkReadiness = "arktos.futurewei.com/network-readiness"
+)
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return new(), nil
+	})
+}
+
+var _ admission.MutationInterface = &plugin{}
+
+type plugin struct {
+	*admission.Handler
+}
+
+// Admit makes an admission decision based on the request attributes
+func (p plugin) Admit(attributes admission.Attributes, o admission.ObjectInterfaces) (err error) {
+	if shouldIgnore(attributes) {
+		return nil
+	}
+
+	pod, ok := attributes.GetObject().(*api.Pod)
+	if !ok {
+		return apierrors.NewBadRequest("Resource was marked with kind Pod but was unable to be converted")
+	}
+
+	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.HostNetwork {
+		return nil
+	}
+
+	if _, readinessExists := pod.Annotations[networkReadiness]; readinessExists {
+		return nil
+	}
+
+	if pod.ObjectMeta.Annotations == nil {
+		pod.ObjectMeta.Annotations = map[string]string{}
+	}
+	pod.ObjectMeta.Annotations[networkReadiness] = "false"
```

Comment:
I believe pod.Annotations(used in lin 61) and pod.ObjectMeta.Annotations (used in line 65,66,68) refer to the same thing. So the code should work, but let's keep the code consistent.

Reply:
they are not. here the annotation is newly added.
a comment will add to make sure it easier to understand